File: check_if_power_of_2.py
#check if a number is a power of 2

def is_Power_of_two(positive_int):
        return positive_int > 0 and (positive_int & (positive_int - 1)) == 0

def solution(positive_int):
    if(positive_int > 0):
        while True:
            if(positive_int % 2 != 1):
                positive_int = positive_int // 2
                if(positive_int == 1):
                    return True 
            else:
                return False


print(is_Power_of_two(4))


print(solution(120))

File: check_if_power_of_2.rb
#check if a number is a power of 2

def solution (positive_int)
    if positive_int > 0
        while true
            if positive_int % 2 != 1
                positive_int = positive_int / 2
                if positive_int == 1
                    return true 
                end
            else
                return false
            end
        end
     end
end


puts(solution(64))